sli/slicontrol.cc

```diff
@@ -1680,51 +1685,32 @@ TimeFunction::execute( SLIInterpreter* i ) const
   i->OStack.push_move( tmp );
 }
 
-/*BeginDocumentation
-Name: sleep_i - suspends proces for n seconds
-Synopsis:  n sleep_i -> -
-Description:
-Calls the POSIX select() function.
-SeeAlso: clock, usertime, tic, toc
-*/
-
-void
-Sleep_iFunction::execute( SLIInterpreter* i ) const
-{
-  i->assert_stack_load( 1 );
-
-  const long sec = static_cast< long >( i->OStack.pick( 0 ) );
-  const long usec = 0;
-  struct timeval tv = { sec, usec };
-  if ( sec > 0 )
-  {
-    select( 0, 0, 0, 0, &tv );
-  }
-
-  i->OStack.pop();
-  i->EStack.pop();
-}
-
-/*BeginDocumentation
-Name: sleep_i - suspends proces for x seconds
-Synopsis:  x sleep_d -> -
-Description:
-Calls the POSIX select() function.
-SeeAlso: clock, usertime, tic, toc
-*/
 
 void
 Sleep_dFunction::execute( SLIInterpreter* i ) const
 {
   i->assert_stack_load( 1 );
 
-  const long sec = 0;
-  const long usec = static_cast< long >(
-    static_cast< double >( i->OStack.pick( 0 ) ) * 1000000. );
+  const double t = static_cast< double >( i->OStack.pick( 0 ) );
 
-  struct timeval tv = { sec, usec };
-  if ( usec > 0 )
+  if ( t < 0 )
+  {
+    throw BadParameterValue( "t >= 0 required." );
+  }
+  else if ( t > std::numeric_limits< int >::max() )
   {
+    throw BadParameterValue( String::compose(
+      "t < %1s required.", std::numeric_limits< int >::max() ) );
```

Comment:
Shouldn't it be `t <=`?

Reply:
The limit is around 68 years (for 32-bit int), so I think it does not really matter much and avoids any potential corner cases. Therefore, I chose the simpler `<`.

Reply:
Alright :+1: